### PR TITLE
Filter out sockets for loopback and unknown interfaces

### PR DESCRIPTION
**Pull request by msherif1234**
## Description

filter out noise flows when RTT and/or pkt drop is enabled for `lo` interface

here is a sample of those waste flows we can see when enable RTT or pkt drop

```sh
{
  "AgentIP": "10.0.122.106",
  "Bytes": 32,
  "DnsErrno": 0,
  "Dscp": 0,
  "DstAddr": "127.0.0.1",
  "DstMac": "00:00:00:00:00:00",
  "DstPort": 80,
  "Duplicate": "false",
  "Etype": 2048,
  "Flags": 512,
  "FlowDirection": "0",
  "IfDirection": 0,
  "Interface": "lo",
  "K8S_FlowLayer": "infra",
  "Packets": 1,
  "Proto": 6,
  "SrcAddr": "127.0.0.1",
  "SrcMac": "00:00:00:00:00:00",
  "SrcPort": 60518,
  "TimeFlowEndMs": 1706717908848,
  "TimeFlowRttNs": 16000,
  "TimeFlowStartMs": 1706717908848,
  "TimeReceived": 1706717913,
  "app": "netobserv-flowcollector"
}

{
  "AgentIP": "10.0.31.177",
  "DnsErrno": 0,
  "Dscp": 0,
  "DstAddr": "::1",
  "DstMac": "00:00:00:00:00:00",
  "DstPort": 80,
  "Duplicate": "false",
  "Etype": 34525,
  "Flags": 2,
  "FlowDirection": "0",
  "IfDirection": 0,
  "Interface": "lo",
  "K8S_FlowLayer": "infra",
  "PktDropBytes": 40,
  "PktDropLatestDropCause": "SKB_DROP_REASON_NO_SOCKET",
  "PktDropLatestFlags": 2,
  "PktDropLatestState": "TCP_INVALID_STATE",
  "PktDropPackets": 1,
  "Proto": 6,
  "SrcAddr": "::1",
  "SrcMac": "00:00:00:00:00:00",
  "SrcPort": 37068,
  "TimeFlowEndMs": 1706722320508,
  "TimeFlowStartMs": 1706722320508,
  "TimeReceived": 1706722320,
  "app": "netobserv-flowcollector"
}
```
## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: bpf/pkt_drops.h [hunk 1]
```diff
@@ -17,7 +17,10 @@ static inline int trace_pkt_drop(void *ctx, u8 state,
     u16 family = 0,flags = 0;
 
     id.if_index = skb->skb_iif;
-
+    // filter out TCP sockets with unknown or loopback interface
+    if (id.if_index == 0 || id.if_index == 1) {
```
Comment (jotak) [inline bpf/pkt_drops.h:21]: is it 100% sure that if_index 0 or 1 is always lo ?

Comment (msherif1234) [reply bpf/pkt_drops.h:21]: yes those two index are reserved , I was hopping if there is api or helper but couldn't find any and saw other app doing the same check to detect lo

**Pull request merged**